charmcraft/linters.py

```diff
@@ -325,10 +347,42 @@ def run(self, basedir: pathlib.Path) -> str:
 
         try:
             with filepath.open("rt", encoding="utf8") as fh:
-                yaml.safe_load(fh)
+                content = dict(yaml.safe_load(fh))
         except Exception:
             return self.Result.ERROR
 
+        actions_names = list(content.keys())
+
+        result_actions, text_actions = check_naming_convention(actions_names, "actions")
+
+        actions_params = [
+            param
+            for action in content.values()
+            if isinstance(action, dict)
+            for param in action.get("params", [])
+        ]
+
+        result_params, text_params = check_naming_convention(actions_params, "action params")
+
+        if result_actions == self.Result.ERROR or result_params == self.Result.ERROR:
```

Comment:
All right! That's fine. I personally feel that people should really not mix naming conventions, but - really - this is your call. Happy to revert to WARNING for everything then.

Reply:
BTW, if all the linting checks should all go into WARNING, we can probably simplify the interface of `check_naming_convention` a bit, with the function to just return True/False. If False we raise warning, otherwise we provide `Result.SUCCESS`

What do you think?

Reply:
From a linter perspective, I agree with you and do consider it an error. Unfortunately, with the current way linters in Charmcraft are set up, an error from a linter prevents completion of the charm build. So `WARNING` is more like a linter error in practice.

What about returning an `Optional[str]` from `check_naming_convention`? Then it can return `None` for success or a warning message for failure.